oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextDocumentMaker.java

```diff
@@ -101,6 +101,8 @@ public FulltextDocumentMaker(@Nullable FulltextBinaryTextExtractor textExtractor
 
     protected abstract void indexTypedProperty(D doc, PropertyState property, String pname, PropertyDefinition pd, int index);
 
+    protected abstract boolean indexDynamicBoost(D doc, PropertyDefinition pd, NodeState nodeState, String propertyName);
```

Comment:
could we maybe change the method name to something else ? Since for elastic, we would actually not be doing index time boosting, and just adding a value that can later on be used for query time boosting.
Maybe supportDynamicBoost or manageDynamicBoost and then add a comment what the method expects from the implementations that extend this class
(Like either do an index time boost or add some property for the boost value which can be used during query time boost )

Reply:
Well all other methods are called "index" so this prefix I would like to keep. "To index" means to add data to the index.

The term "dynamic boost": I called the feature that way. I think the term should reflect what it is doing on a high level, and not how it's implemented. An alternative: we could use "dynamic confidence", if "boost" is too much a Lucene / Elasticsearch term. But I don't think "confidence" is better than "boost"... It is about boosting some documents. Whether that's done at indexing time or query time is an implementation detail.

So "indexDynamicBoost" means: Store the data in the index that is needed for the "dynamic boost" feature.